### PR TITLE
have some problems

### PR DESCRIPTION
**Pull request by purpleFade**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,61 +1,118 @@
...
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const todos = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId)
+  || {
+    id: 0, name: '', username: '', email: '',
+  },
+})) as TodoWithUser[];
```
Comment (etojeDenys) [inline src/App.tsx:15]: it's redundant let's just return null

File: src/App.tsx [hunk 2]
```diff
@@ -1,61 +1,118 @@
...
 export const App = () => {
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+
+  const [posts, setPosts] = useState<TodoWithUser[]>(todos);
+
+  const addPost = (post: TodoWithUser) => {
```
Comment (etojeDenys) [inline src/App.tsx:26]: image

File: src/App.tsx [hunk 3]
```diff
@@ -1,61 +1,118 @@
...
+    e.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    const newPost: TodoWithUser = ({
```
Comment (etojeDenys) [inline src/App.tsx:48]: image

File: src/components/TodoInfo/TodoInfo.tsx [hunk 4]
```diff
@@ -1 +1,18 @@
...
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const { title, completed, user } = todo;
+
+  return (
+    <article className={`TodoInfo ${completed ? 'TodoInfo--completed' : ''}`}>
+      <h2 className="TodoInfo__title">{title}</h2>
+
+      <UserInfo user={user} />
```
Comment (etojeDenys) [inline src/components/TodoInfo/TodoInfo.tsx:15]: user can be null let's handle this case here

**Review by etojeDenys — CHANGES_REQUESTED** [review]

**Review by lerastarynets — CHANGES_REQUESTED** [review]
Did you forget to push changes?) Previous comments aren’t fixed
Also pls add demo link

**Comment by purpleFade** [comment]
[DEMO LINK](https://purplefade.github.io/react_add-todo-form/)

File: src/App.tsx (hunk 1 above)
Comment (IvanFesenko) [reply src/App.tsx:15]: Not fixed 
2023-09-05_12h00_54
```suggestion
  user: getUserById(todo.userId)
}));
```

File: src/App.tsx [hunk 5]
```diff
@@ -1,61 +1,126 @@
...
+      id: posts.length + 1,
+      title,
+      completed: false,
+      userId,
+      user: getUserById(userId) || {
+        id: 0,
+        name: '',
+        username: '',
+        email: '',
+      },
```
Comment (IvanFesenko) [inline src/App.tsx:58]:
```suggestion
      user: getUserById(userId),
```

File: src/components/TodoInfo/TodoInfo.tsx [hunk 6]
```diff
@@ -1 +1,18 @@
...
+
+type Props = {
+  todo: TodoWithUser;
+};
+
+export const TodoInfo: React.FC<Props> = ({ todo }) => {
+  const { title, completed, user } = todo;
+
+  return (
+    <article className={`TodoInfo ${completed ? 'TodoInfo--completed' : ''}`}>
```
Comment (IvanFesenko) [inline src/components/TodoInfo/TodoInfo.tsx:12]: Not fixed
2023-09-05_12h03_51

File: src/components/TodoInfo/TodoInfo.tsx [hunk 7]
```diff
@@ -1 +1,18 @@
...
+      <h2 className="TodoInfo__title">{title}</h2>
+
+      {user !== undefined && <UserInfo user={user} />}
```
Comment (IvanFesenko) [inline src/components/TodoInfo/TodoInfo.tsx:15]:
```suggestion
      {user && <UserInfo user={user} />}
```

File: src/types/types.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+};
+
+export type Users = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
+
+export type TodoWithUser = Todos & { user?: Users };
```
Comment (IvanFesenko) [inline src/types/types.ts:15]:
```suggestion
export type TodoWithUser = Todos & { user: Users | null };
```

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Not all previous comments were fixed. Check them below

**Pull request closed without merging**